advanced/Scripts/query.sh

```diff
@@ -58,12 +61,68 @@ scanList(){
     esac
 }
 
+# function to count new domain entries since last gravity run
+CountNewDomains() {
+  newdomainentries=$(
+    sqlite3 "${gravityDBfile}" << EOSQL
+      ATTACH '${gravityOLDfile}' AS old;
+      SELECT count(DISTINCT domain) FROM gravity WHERE domain NOT IN (SELECT domain FROM old.gravity);
+EOSQL
+      )
+  echo -e "  ${INFO} ${newdomainentries} new domain(s) found."
+  exit 0
+}
+
+# function to find blocked entries in the FTL database,
+# present (new) in the latest (active) gravity database,
+# and NOT present in the old gravity database.
+FindNewBlockedDomains() {
+  # Check if the OLD database exists.
+  if [[ ! -f "${gravityOLDfile}" ]]; then
+    echo -e "  ${CROSS}  Old database ${gravityOLDfile} doesn't exist."
+    exit 1
+  else
+    # Check if the current database is more recent.
+    if [[ $(sqlite3 "${gravityOLDfile}" "SELECT value FROM info \
+              WHERE property IS 'updated';") \
+        > \
+          $(sqlite3 "${gravityDBfile}" "SELECT value FROM info \
+              WHERE property IS 'updated';") ]]; then
+      echo -e "  ${CROSS}  Old database ${gravityOLDfile} is more recent."
+      exit 1
+    fi
+    #Run the query
+    mapfile -t resultarray < <( sqlite3 -separator ' ' "${ftlDBfile}" << EOSQL
+      ATTACH '${gravityDBfile}' AS new;
+      ATTACH '${gravityOLDfile}' AS old;
+      SELECT DISTINCT queries.domain, '(' || adlist.address || ')' FROM queries
+      JOIN gravity ON new.gravity.domain = queries.domain
+      JOIN adlist ON adlist.id = new.gravity.adlist_id
+      WHERE queries.domain NOT IN (SELECT domain FROM old.gravity)
+      AND queries.domain IN (SELECT domain FROM new.gravity)
+      AND queries.status IN (1,9)
```

Comment:
In case of type 9, I think you need to take the domain from `additional_info` column because this is the actual blocked domain.

You might want to separate the CNAME stuff as it might get a bit to complexe otherwise.

Reply:
Would this be acceptable for you?
`pi@raspberrypi:~ $ pihole -q -blocked
  [i] Blocked: amzonse.xyz (https://phishing.army/download/phishing_army_blocklist_extended.txt)
  [i] Blocked: aod.cm.cn.miaozhen.com (https://www.github.developerdan.com/hosts/lists/ads-and-tracking-extended.txt)
  [i] Blocked: www.dnsimple.com. (CNAME)
        Domain: dnsimple.com (file:///home/pi/test.txt)
`
![image](https://user-images.githubusercontent.com/24669386/118401450-16c71f80-b666-11eb-8ea2-54a619480e13.png)

Reply:
Which domains is which? `www.dnsimple.com` is CNAME for `dnsimple.com`?

It wasn't meant as a hint for how to display things, but actually to include  domains from `additional_info` into your comparison. (You might have addressed this already and not just the output)

Reply:
I added a test list (could be a domain in a regular list) dnsimple.com. I than ran a dig www.dnsimple.com (not in any list). Since this query triggers CNAME detection (status 9), the separate query (as you suggested) finds www.dnsimple.com with status 9, so the user is informed the dns query for this domain, which is a CNAME was blocked. the second line (one for each CNAME) lists the gravity entry (dnsimple.com) and the list.

I think this should be very clear to the user, the output matches the info in the query log

![image](https://user-images.githubusercontent.com/24669386/118406829-e9866b80-b67d-11eb-8c63-4d0db4a0b91b.png)

Reply:
I've not checked the query in detail, but from a glance it looked like what I had in mind.